pyuvdata/uvflag/uvflag.py

```diff
@@ -1286,6 +1589,64 @@ def to_baseline(
                 '"baseline" to match.'
             )
 
+        # write it out this rather than comparing the UVParameters because
+        # future_array_shapes might be different. In the future, when shapes are not
+        # variable, this can be done by comparing the UVParameters.
+        if self.Nfreqs != uv.Nfreqs or not np.allclose(
+            np.squeeze(self.freq_array),
+            np.squeeze(uv.freq_array),
+            rtol=self._freq_array.tols[0],
+            atol=self._freq_array.tols[1],
+        ):
+            raise ValueError(
+                "The freq_array on uv is not the same as the freq_array on this "
+                f"object. The value on this object is {self.freq_array}; the value "
+                f"on uv is {uv.freq_array}"
+            )
+
+        warn_compatibility_params = [
+            "telescope_name",
+            "telescope_location",
+            "antenna_names",
+            "antenna_numbers",
+            "antenna_positions",
+            "channel_width",
+            "spw_array",
+        ]
+        if self.Nspws > 1:
+            warn_compatibility_params.append("flex_spw_id_array")
```

Comment:
I think it should error. But I didn't want to create new errors without a warning period because this would not have errored in the past. (A user could get this error on code they have run successfully before because the way UVFlag objects are usually initialized is from UVData or UVCal objects, and that initialization now sets the `flex_spw_id_array` if it exists on the initial object).

Reply:
That's totally fair -- maybe worth adding a TODO here just so that we don't forget?

Reply:
I think that's already there -- the warning this triggers is a deprecation warning that says it will become an error in version 2.4